### ISSUE TITLE
Including last=false in non-last metrics

### ISSUE THREAD

**Issue by jakubgs** [mgumz/mtr-exporter#12]
Hello. I really like the `0.3.0` release with the support for multiple jobs and flag for legacy metrics names. Very cool.

I was trying out the new release and I'm having trouble using the `last` label in Grafana. Maybe you have some ideas.
When I try to just do a constant variable that can be either `true` or `false`, and then a query:
```
mtr_report_loss{mtr_exporter_job=~"[[job:pipe]]", last="${last}"}
```
It works well then `last` is `true`, but if I switch `last` to be `false` I instead get nothing. And this is because only last metrics get `last=true`, but other metrics do not get `last=false`. This makes it awkward to use this label.

What do you think about including `last=false` in other metrics?

**Comment by jakubgs** [comment]
I can work around this by setting `last` variable to an empty string:
```
mtr_report_loss{mtr_exporter_job=~"[[job:pipe]]", last=""}
```
And that does give me metrics that do not have the `last` label, but this is impossible to use with Grafana, because the variable doesn't support an empty string option:

![image](https://github.com/mgumz/mtr-exporter/assets/2212681/d55e1563-9249-4b30-8ac7-3759ab4356e4)

It only recognizes a string that non-zero in length.

**Comment by jakubgs** [comment]
This should be pretty simple to implement based on this:
https://github.com/mgumz/mtr-exporter/blob/7b230d0e30de64a30b6431e806c94de73e888f5e/pkg/job/prometheus.go#L85-L92
Just asking if you agree. Happy to make a PR.

**Comment by mgumz** [comment]
I agree, to add "last" to all the metrics is easy.

Main question is: why? What kind of result to you show when you have `last="false"`? You see all all but the last hop.

Don't get me wrong, I am not against the idea. I am currently thinking of the downside of that approach. Another related idea is to label the first one as "first" … and the hops in the middle as "intermediate". But then one would have to also apply the same logic like `intermediate="false"` to the first and the last. And this smells kind of ugly.

With `first` and `last` and `intermediate` you would be able to actually select from a combo-box. What do you think about this twist of the idea?

**Comment by jakubgs** [comment]
>Main question is: why? What kind of result to you show when you have last="false"? You see all all but the last hop.

No, that's incorrect. I see nothing:

![image](https://github.com/mgumz/mtr-exporter/assets/2212681/11fbb9f7-f0e8-4595-bddd-fc18e70eaba9)

**Comment by jakubgs** [comment]
>With `first` and `last` and `intermediate` you would be able to actually select from a combo-box. What do you think about this twist of the idea?

I like this idea, but what if it was just one field? You could call it `hop_type` for example(or something like that), and the possible values would be `first`/`last`/`intermediate`.

**Comment by mgumz** [comment]
one field: yah, but there is the situation when `first` is also `last` :)

check https://github.com/mgumz/mtr-exporter/pull/13 … give it a try

**Comment by jakubgs** [comment]
>one field: yah, but there is the situation when first is also last :)

Hmmm, that's a fair point, but an edge case. As always edge cases ruin everything.
Honestly, if it was me, I'd just call that `hop_type=all`. Because handling 3 different lables would be annoying in Grafana.
But I understand you argument, it makes sense.

**Comment by mgumz** [comment]
> > Main question is: why? What kind of result to you show when you have last="false"? You see all all but the last hop.
> 
> No, that's incorrect. I see nothing:
> 
> ![image](https://user-images.githubusercontent.com/2212681/266776686-11fbb9f7-f0e8-4595-bddd-fc18e70eaba9.png)

Yah, my mistake: Lets assume I have implemented `laste="false"` … then you would see all hops BUT the last :) And I was curious on what this is actually good for. Atm, yes, no data =)

**Comment by jakubgs** [comment]
> And I was curious on what this is actually good for.

It's good for seeing the health of intermediate devices without seeing the final destination. For example number of ICMP packets dropped by intermediate routers is a proxy for how overloaded they are, since they usually have ICMP as lowest priority traffic, and the more load they have the more they drop.

**Comment by mgumz** [comment]
Another approach … you can use a Prometheus recording rule … and just add `last="false"` where `last` is not "true" already.

**Comment by jakubgs** [comment]
That's true, you can adjust labels via Prometheus, but honestly, that's just a crutch and not a real fix.

**Comment by mgumz** [comment]
Its a fix to your scratch =)

> It's good for seeing the health of intermediate devices without seeing the final destination.

True. I usually tend to ignore the intermediates … but in an saturation situation, when the intermediates are as equally bad as the last one … you can see really the issue. And then a query against intermediate vs last makes sense.

I ll think about the the feature, also the `hop_type`. `hop_type="first_last"` would also be do it. It would show up on `hop_type=~"first"` and `hop_type=~"last"`

**Comment by mgumz** [comment]
i changed #13 a bit to outline the `hubType` label idea. If you want to toy around with it: be my guest.

**Comment by jakubgs** [comment]
>I ll think about the the feature, also the hop_type. `hop_type="first_last"` would also be do it. It would show up on `hop_type=~"first"` and `hop_type=~"last"`.

That's a very good point actually. It plays well with a regex, clever.

**Comment by mgumz** [comment]
Implemented via #13
